### PR TITLE
fix: API "crashing"

### PR DESCRIPTION
**Pull request by Skribb11es**
# The Fix

Removed the headers, and in one case added a check for the current header being used, which were causing the issue.

```py
"Origin": "https://lolesports.com", "Referrer": "https://lolesports.com"
```

This, as far as I can tell, completely fixes the issue with Riot's servers being "overloaded" 🤦‍♂️ (nobody wanted to believe me), while also being the simplest possible fix. In the event that this solution is not adequate, I have devised another that does not remove the headers from the request and instead changes the headers accordingly to the current state of Riot's API, assuming these lines had any use of course.

# Why I Think It Was Happening 

In my eyes, and this is complete speculation, Riot is either manually changing the required headers for the request to the drop API in an attempt to mitigate traffic, or they have their APIs configured to automatically change to a different server. However, it appears the setup for the other server uses `Referer` instead of `Referrer` as the header, causing the farmer to break when Riot's servers change. 

Credit to `RustyWalls#9782`, `Charlie#1001`, and `dimitrisz123#1466` for helping me test the changes, and giving feedback & ideas.

# How To Install

You will not receive any support for either of these install methods, these are completely band-aid fixes while we wait for Poro to merge and make an official build.

## Normal Users

If you use the built, .exe, version of the farmer please download Penguax's build [here](https://github.com/Penguax/CapsuleFarmerEvolved1/releases/tag/fix3)

## Advanced Users

1. If you are running an advanced install, simply clone the branch by running `git clone --branch API-fix https://github.com/Skribb11es/CapsuleFarmerEvolved.git` in your terminal of choice (this should ideally be done in a folder that does not already have a pre-existing install of the farmer). 
2. Once you have cloned the branch, move your config file from your previous install to the new install and run the program normally.
3. That's it, yep it's really that easy...

### CODE REVIEW

**Comment by Skribb11es** [comment]
Really had to make 3 PR's and finally have Rusty make a new branch just for the DIFF to be correct, ffs

**Comment by Penguax** [comment]
Lots of testing talk in #dev channel on Discord. 

# SKIP images for summary

Below is some testing with the fixes in this PR 
### 1.Both Farmers are identical except for `accountname` 
image
### 2. Changed `GeoLocation` on the RIGHT farmer
image
### 3. Changed `GeoLocation` + `Referer` fix changes on RIGHT farmer.
image
### 4. Changed ONLY `Referer` on RIGHT farmer. 
image

Summary: 
Changing geolocation didn't affect the login / "api" status of the farmer, however changing `Referer` did, I can also confirm ive already gotten 3 drops using the fix. 

*Note* 
Changing `Referrer` -> `Referer`, also fixes this issue, however completely removing the referer lines also negates the issue of having to implement a fix to flip flop between the two spellings in the code to satisfy the wanted request headers.

**Comment by auroraisluna** [comment]
Please also update your PR to include the standard PR template: https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by Skribb11es** [comment]
> Please also update your PR to include the standard PR template: https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/blob/master/.github/PULL_REQUEST_TEMPLATE.md

As is, it includes everything from the "standard PR template" just using different wording, it's even laid out in the exact same order.

**Comment by auroraisluna** [comment]
> > Please also update your PR to include the standard PR template: https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/blob/master/.github/PULL_REQUEST_TEMPLATE.md
> 
> As is, it includes everything from the "standard PR template" just using different wording, it's even laid out in the exact same order.

The template is there to be followed to the letter. Please update your PR description to use the template. Suggesting moving the first content of your current under "Summary", everything else under additional context 👍
Key Missing Sections: 
- [ ] 'Testing instructions' section
- [ ] 'How to download the PR for testing' required section 

Thanks.

File: src/main.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from Stats import Stats
 from VersionManager import VersionManager
 
-CURRENT_VERSION = 1.3
+CURRENT_VERSION = 1.33
```
Comment (LeagueOfPoro) [inline src/main.py:20]: Don't change the version

File: src/Browser.py [hunk 2]
```diff
@@ -97,16 +98,23 @@ def login(self, username: str, password: str, refreshLock) -> bool:
             self.client.get(
                 "https://auth.riotgames.com/authorize?client_id=esports-rna-prod&redirect_uri=https://account.rewards.lolesports.com/v1/session/oauth-callback&response_type=code&scope=openid&prompt=none&state=https://lolesports.com/?memento=na.en_GB", allow_redirects=True).close()
 
-            # Get access and entitlement tokens for the first time
-            headers = {"Origin": "https://lolesports.com",
-                        "Referrer": "https://lolesports.com"}
+            def reqAcc():
+                # This requests sometimes returns 404
+                return self.client.get(
+                    "https://account.rewards.lolesports.com/v1/session/token", headers={"Origin": "https://lolesports.com", self.ref: "https://lolesports.com"})
```
Comment (LeagueOfPoro) [inline src/Browser.py:104]: Don't create a nested function. Looking at @Penguax testing, is the function even necessary? Could we only remove te Referer completely?

File: src/main.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def init() -> tuple[logging.Logger, Config]:
     args = parser.parse_args()
 
     print("*********************************************************")
-    print(f"*   Thank you for using Capsule Farmer Evolved v{CURRENT_VERSION}!    *")
+    print(f"*   Thank you for using Capsule Farmer Evolved v{str(CURRENT_VERSION)}!    *")
```
Comment (LeagueOfPoro) [inline src/main.py:30]: the str is not needed

**Review by LeagueOfPoro — CHANGES_REQUESTED** [review]

**Comment by LeagueOfPoro** [comment]
I currently don't have access to pc to test it. It looks like @Penguax did. Can we just remove referer everywhere?

File: src/main.py (hunk 3 above)
Comment (Skribb11es) [reply src/main.py:30]: resolved the coloring issue seen with the version number being half white and half blue in the console.

File: src/Browser.py (hunk 2 above)
Comment (Skribb11es) [reply src/Browser.py:104]: Explained in discord. #capsule-farmer-dev [here](https://discord.com/channels/493691955184926720/1071031553460817990/1076948085827194890)

**Comment by Skribb11es** [comment]
> It looks like @Penguax did. I tried it and I think we just remove referer everywhere.

explained via discord, and also removed the nested func.

**Comment by Penguax** [comment]
removing `referer` everywhere caused some login issues for some users/ when adding new accounts afterwords. when only removing it from the `watch` area it worked fine for them, but there's no knowing if in the future they also start denying watch packets with missing referer headers.

**Comment by Skribb11es** [comment]
Well more specifically adding to the above comment, Riot's API servers switch their headers back and forth from time to time, presumably based on load, as I said in the PR's initial comment. However, this does also mean that each individual API has its own headers, which does complicate things but is also pretty easy to catch if you simply add an if elif statement as I did for "v 1.33".

**Review by LeagueOfPoro — APPROVED** [review]

**Pull request merged**